lib/rebay_api_web/controllers/item_controller.ex

```diff
@@ -1,43 +1,83 @@
 defmodule RebayApiWeb.ItemController do
   use RebayApiWeb, :controller
 
+  alias RebayApi.Accounts
   alias RebayApi.Listings
   alias RebayApi.Listings.Item
 
   action_fallback RebayApiWeb.FallbackController
 
+  plug :authenticate_session when action in [:create, :update, :delete]
+  plug :authorize_user when action in [:create, :update, :delete]
+
   def index(conn, _params) do
     items = Listings.list_items()
     render(conn, "index.json", items: items)
   end
 
-  def create(conn, %{"item" => item_params}) do
-    with {:ok, %Item{} = item} <- Listings.create_item(item_params) do
+  def create(conn, _params) do
+    user = conn.params["user_uuid"] |> Accounts.get_user!()
+
+    create_attrs = Map.put(conn.params, "user_id", user.id)
+    |> Map.put("uuid", Ecto.UUID.generate)
+
+    with {:ok, %Item{} = item} <- Listings.create_item(create_attrs) do
       conn
       |> put_status(:created)
-      |> put_resp_header("location", Routes.item_path(conn, :show, item))
+      |> put_resp_header("location", Routes.user_item_path(conn, :show, user.uuid, item))
       |> render("show.json", item: item)
     end
   end
 
-  def show(conn, %{"id" => id}) do
-    item = Listings.get_item!(id)
+  def show(conn, %{"uuid" => uuid, "user_uuid" => _user_uuid}) do
+    item = Listings.get_item!(uuid)
     render(conn, "show.json", item: item)
   end
 
-  def update(conn, %{"id" => id, "item" => item_params}) do
-    item = Listings.get_item!(id)
+  def update(conn, %{"uuid" => uuid, "item" => item_params}) do
+    item = Listings.get_item!(uuid)
 
     with {:ok, %Item{} = item} <- Listings.update_item(item, item_params) do
       render(conn, "show.json", item: item)
     end
   end
 
-  def delete(conn, %{"id" => id}) do
-    item = Listings.get_item!(id)
+  def delete(conn, %{"uuid" => uuid, "user_uuid" => _user_uuid}) do
```

Comment:
It doesn't look like there's anything here that would prevent me from modifying someone else's item, as long as I'm logged in... is that true?

Reply:
Yes, we actually just noticed this while testing frontend integration!

Reply:
This is resolved in PR #10